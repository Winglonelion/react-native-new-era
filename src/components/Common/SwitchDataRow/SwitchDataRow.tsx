import React, { memo } from 'react';
import { StyleSheet, Switch, View } from 'react-native';
import ContentText from 'components/Base/Text/ContentText';
import Colors from 'theme/colors';

interface PropTypes {
  title: string;
  onChange?: (value: boolean) => void;
  isEnabled: boolean;
}

const SwitchDataRow: React.FC<PropTypes> = memo(
  ({ title, onChange, isEnabled = true }) => {
    return (
      <View style={styles.container}>
        <View style={styles.content}>
          <ContentText size={16} style={styles.text}>
            {title}
          </ContentText>
          {!!onChange && (
            <Switch
              trackColor={{ false: Colors.grey, true: Colors.orange }}
              thumbColor={isEnabled ? Colors.white : '#f4f3f4'}
              ios_backgroundColor={Colors.dark}
              onValueChange={onChange}
              value={isEnabled}
            />
          )}
        </View>
      </View>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    paddingHorizontal: 15,
    alignItems: 'center',
  },
  content: {
    paddingVertical: 15,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    flex: 1,
  },
  contentTxt: {
    color: Colors.lightBrown,
  },
  text: {
    lineHeight: 30,
  },
});

export default SwitchDataRow;
