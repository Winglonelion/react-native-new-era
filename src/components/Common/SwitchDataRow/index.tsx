export { default as default } from './SwitchDataRow';
