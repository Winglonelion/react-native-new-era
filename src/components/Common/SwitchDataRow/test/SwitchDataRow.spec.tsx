import React from 'react';
import { shallow } from 'enzyme';
import SwitchDataRow, { PropTypes } from '../SwitchDataRow';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<SwitchDataRow {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('SwitchDataRow check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      title: 'switch title',
      isEnabled: true,
      onChange: jest.fn(),
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot isEnabled false', () => {
    const { wrapper } = setup({
      title: 'switch title 2',
      isEnabled: false,
      onChange: jest.fn(),
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot without onChange', () => {
    const { wrapper } = setup({
      title: 'switch title 3',
      isEnabled: false,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
