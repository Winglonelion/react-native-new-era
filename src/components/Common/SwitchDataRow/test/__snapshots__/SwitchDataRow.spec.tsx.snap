// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchDataRow check snapshot should match snapshot isEnabled false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 15,
      }
    }
  >
    <Memo()
      size={16}
      style={
        Object {
          "lineHeight": 30,
        }
      }
    >
      switch title 2
    </Memo()>
    <Switch
      ios_backgroundColor="#373F41"
      onValueChange={[MockFunction]}
      thumbColor="#f4f3f4"
      trackColor={
        Object {
          "false": "#DADFE6",
          "true": "#FF894D",
        }
      }
      value={false}
    />
  </View>
</View>
`;

exports[`SwitchDataRow check snapshot should match snapshot with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 15,
      }
    }
  >
    <Memo()
      size={16}
      style={
        Object {
          "lineHeight": 30,
        }
      }
    >
      switch title
    </Memo()>
    <Switch
      ios_backgroundColor="#373F41"
      onValueChange={[MockFunction]}
      thumbColor="#FFFFFF"
      trackColor={
        Object {
          "false": "#DADFE6",
          "true": "#FF894D",
        }
      }
      value={true}
    />
  </View>
</View>
`;

exports[`SwitchDataRow check snapshot should match snapshot without onChange 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 15,
      }
    }
  >
    <Memo()
      size={16}
      style={
        Object {
          "lineHeight": 30,
        }
      }
    >
      switch title 3
    </Memo()>
  </View>
</View>
`;
